src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -204,6 +204,29 @@ internal static class CommonEnvVariableNames
 #endif
         }
 
+        /// <summary>
+        /// Get a temporary directory to use, needs to be unique to avoid collision
+        /// </summary>
+        public static string GetTemporaryDirectory()
+        {
+            string tempDir = "";
+            do
+            {
+                tempDir = Path.Combine(Path.GetTempPath(),System.Guid.NewGuid().ToString());
```

Comment:
It doesn't really matter here because the loop should only run once, but as a matter of style, you should move calls that always (or should always) return the same thing (like `Path.GetTempPath`) outside of the loop.